### ISSUE TITLE
move logic to middleware/transforms

### ISSUE THREAD

**Issue by jonschlinkert** [jonschlinkert/template#18]
@doowb I started moving some of the init/default config stuff into transforms. I think there is probably a lot of stuff we're doing in template that could be moved into built-in middleware or transforms. initially to promote modularity and simplify some of the logic, but it will also make it easier to find places to either move logic to user land, remove stuff, or find code that can be generalized.

thoughts?

**Comment by doowb** [comment]
I agree... I was thinking the same thing. There are a couple of utility functions I was thinking of too that would help create the default configurations. I like using transforms for these things because it'll clean up the main file a lot.
